smf/parser.js

```diff
@@ -41,7 +41,7 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     if (match[8] == 'php') {
       result.rtype = 'REF';
     }
-  } else if ((match = /^(\/en)?\/[a-z]+\/([a-z\S+]+)\//i.exec(path)) !== null) {
+  } else if ((match = /^(\/en)?\/[a-z]+\/([a-z\S+]+)\/([0-9\/]+)?([a-z\/]+)?/i.exec(path)) !== null) {
```

Comment:
This expression is pretty strange : `([a-z\S+]+)`
`\S` includes `[a-z]` as well as `+` (yep, here `+` is a character, not an operator)
Be careful when you includes **slashes** into your expressions, this can cause edge effects. For example, if the URL ends with something like `/aaa/111/aaa`, `\S+` will catch `aaa/111`.

More concrete example :
```javascript
  let reg = /^(\/en)?\/[a-z]+\/([a-z\S+]+)\/([0-9\/]+)?([a-z\/]+)?/;
  let match = reg.exec('/en/Publications/Bulletin/144/html/');
  console.log(match[2]); // Publications/Bulletin/144/html
```